extensions/scarb-doc/src/docs_generation/markdown/traits.rs

```diff
@@ -215,3 +212,30 @@ fn generate_markdown_from_item_data(doc_item: &dyn DocItem, header_level: usize)
 
     markdown
 }
+
+pub trait TopLevelMarkdownItemsSummary {
+    #[allow(unused)]
+    fn summary_filename(&self) -> &str;
+}
+
+macro_rules! impl_top_level_markdown_items_summary {
+    ($t:ty, $filename:expr) => {
+        impl TopLevelMarkdownItemsSummary for Vec<&$t> {
+            fn summary_filename(&self) -> &str {
+                $filename
+            }
+        }
+    };
+}
```

Comment:
Same idea as above, I will need the file names in `SUMMARY.md`, so I want to keep them hardcoded in one place